packages/dd-trace/src/appsec/iast/taint-tracking/plugin.js

```diff
@@ -170,6 +181,37 @@ class TaintTrackingPlugin extends SourceIastPlugin {
     this.taintHeaders(req.headers, iastContext)
     this.taintUrl(req, iastContext)
   }
+
+  _taintDatabaseResult (result, dbOrigin, iastContext = getIastContext(storage.getStore()), name) {
+    if (!iastContext) return result
+
+    let rowsToTaint = this.iastConfig?.dbRowsToTaint
+    if (typeof rowsToTaint !== 'number') {
+      rowsToTaint = 1
+    }
+
+    if (rowsToTaint === 0) return
+
+    if (Array.isArray(result)) {
+      for (let i = 0; i < result.length && i < rowsToTaint; i++) {
+        const nextName = name ? `${name}.${i}` : '' + i
+        result[i] = this._taintDatabaseResult(result[i], dbOrigin, iastContext, nextName)
+      }
+    } else if (result && typeof result === 'object') {
+      if (dbOrigin === 'sequelize' && result.dataValues) {
```

Comment:
IMO distinguishing between databases using `dbOrigin` in this function is unnecessary and could make the code more complex if we add more database subscriptions in the future.

I suggest sending `result.dataValues` directly from the sequelize subscription and removing this part from the function. wdyt?

Reply:
It was my first approach, but it will be super expensive. If we are going to taint only one row, we don't need to iterate each row of the data just to taint only the first one. And this object could be complex, like 100 rows of:

```javascript
[
  { ..., dataValues: { id: 123, name: "name", children: [{.., dataValues: { /*another complex object */ }]}
]
```

This is the reason why I decided to check the database type here and not in the instrumentation, going against our good practices.